### ISSUE TITLE
the content-type of svg image file is crashed after gzip compressing sync to aws

### ISSUE THREAD

**Issue by hothero** [AssetSync/asset_sync#323]

**Comment by danielfarrell** [comment]
For those who find this later, this seems to be better handled by registering the svg mime type in an initializer:

```
Mime::Type.register "image/svg+xml", :svg
```

**Comment by coneybeare** [comment]
@danielfarrell This didn't work for me when placed in `mime_types.rb`. Are you successfully preventing `asset_sync` from compressing svg's upon upload?

**Comment by danielfarrell** [comment]
I wasn't trying to prevent compression. It should be fine if they are
compressed as long as the mime type gets set properly on S3. Without this
the mime type was getting set to application/gzip for me. Maybe we are
actually after different end results.

**Comment by coneybeare** [comment]
Nope, I was misidentifying the problem. Was set as txt/plain before and your fix made it work, once my cloudfront copy was invalidated (face-palm)

**Comment by PikachuEXE** [comment]
@coneybeare 
Do you think #322 is still required?

**Comment by hothero** [comment]
So we just register Mime::Type "image/svg+xml" and will not get nil from Mime::Type.lookup_by_extension(ext), right?

If true, maybe readme need to append more advanced usage. But I think #322 still have some use for the future, Mime::Type looks have no all of types.

**Comment by PikachuEXE** [comment]
So what is a preferred API for searching mime type?
`::MIME::Types.type_for` or `Mime::Type.lookup_by_extension`?

**Comment by coneybeare** [comment]
I am not familiar enough in this area to have a strong opinion, but I am generally in favor of things that work "out of the box" without any additional manual configuration additions.

**Comment by PikachuEXE** [comment]
@coneybeare 
I am fine with that
But I am not familiarized with that mime types gem

**Comment by danielfarrell** [comment]
Seems to me the best way to handle this is a PR against that gem to add the missing mime types.  Anyone have a list of ones that are commonly issues?

**Comment by PikachuEXE** [comment]
@danielfarrell 
I am more interested in why the result give by
- `:MIME::Types.type_for`
- `Mime::Type.lookup_by_extension`
are different
I don't know where are these 2 methods documented
So it *could* be our usage is incorrect

**Comment by lloydwatkin** [comment]
I'm seeing the same, my **svg** files get to cloudfront (AWS CDN) as `application/gzip` which means they aren't displayed. Work around is **png** for now, but would prefer **svg**.

**Comment by lime** [comment]
@lloydwatkin Have you tried the suggestion above to register the right mime type for svg?

```ruby
# config/initializers/mime_types.rb
Mime::Type.register "image/svg+xml", :svg
```

Rails recently got the above added to the default mime types: https://github.com/rails/rails/commit/7e35cb2987e146de29d50f97a43caef61e7588af It's available in 5.0.0.

**Comment by lloydwatkin** [comment]
Yes but it complained it was already defined and I'm on rails 4 at the
moment. Will investigate when I get a chance. Thanks!
On 5 Jul 2016 12:38, "Emil Sågfors" <notifications@github.com> wrote:

@lloydwatkin <https://github.com/lloydwatkin> Have you tried the suggestion
above to register the right mime type for svg?

# config/initializers/mime_types.rbMime::Type.register "image/svg+xml", :svg

Rails recently got the above added to the default mime types: rails/rails@
7e35cb2
<https://github.com/rails/rails/commit/7e35cb2987e146de29d50f97a43caef61e7588af>
It's available in 5.0.0.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/AssetSync/asset_sync/issues/323#issuecomment-230455358>,
or mute the thread
<https://github.com/notifications/unsubscribe/AAQlBu3ebLp5DMCrnfSZeWRIuXs2WQvEks5qSkJRgaJpZM4Hik-t>
.

**Comment by PikachuEXE** [comment]
I guess we should let users know they need to put something like
`Mime::Type.register "image/svg+xml", :svg` for custom file types
And also let them know there are some types already registered by rails (and that list might change in the future)
